internal/sqlsmith: TestGenerateParse failed

[(internal/sqlsmith).TestGenerateParse failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2781103&tab=buildLog) on [master@ca60935df431525717592574475f34b5ed422ea9](https://github.com/cockroachdb/cockroach/commits/ca60935df431525717592574475f34b5ed422ea9):

```
I210316 05:59:06.200390 7777 sql/temporary_schema.go:492  [n1] 494  running temporary object cleanup background job
I210316 05:59:06.204964 7775 1@util/log/event_log.go:32  [n1] 495 ={"Timestamp":1615874346197665647,"EventType":"node_join","NodeID":1,"StartedAt":1615874346190475964,"LastUp":1615874346190475964}
I210316 05:59:06.208144 7803 sql/sqlliveness/slstorage/slstorage.go:352  [n1] 496  inserted sqlliveness session e3526785cb174d0c9a09198ad4fac7db
I210316 05:59:06.208243 7803 sql/sqlliveness/slinstance/slinstance.go:144  [n1] 497  created new SQL liveness session e3526785cb174d0c9a09198ad4fac7db
I210316 05:59:06.211626 7784 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 498 ={"Timestamp":1615874346205835141,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I210316 05:59:06.217147 7698 3@vendor/github.com/cockroachdb/pebble/db.go:1449  [n1,pebble] 499  [JOB 3] WAL created 000004
I210316 05:59:06.217411 8034 migration/migrationmanager/manager.go:102  [n1,intExec=set-setting,migration-mgr] 500  no need to migrate, cluster already at newest version
I210316 05:59:06.218996 7777 sql/temporary_schema.go:527  [n1] 501  found 0 temporary schemas
I210316 05:59:06.219052 7777 sql/temporary_schema.go:530  [n1] 502  early exiting temporary schema cleaner as no temporary schemas were found
I210316 05:59:06.219085 7777 sql/temporary_schema.go:531  [n1] 503  completed temporary object cleanup job
I210316 05:59:06.219105 7777 sql/temporary_schema.go:610  [n1] 504  temporary object cleaner next scheduled to run at 2021-03-16 06:29:06.200377543 +0000 UTC
I210316 05:59:06.220486 8034 util/log/event_log.go:32  [n1,intExec=set-setting] 505 ={"Timestamp":1615874346214455533,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'20.2-48'"],"SettingName":"version","Value":"20.2-48"}
I210316 05:59:06.250926 8373 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 506 ={"Timestamp":1615874346247291189,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"da4f595d-7216-4d08-9fdd-7d890d6ee6ca"}
I210316 05:59:06.254835 8348 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 507 ={"Timestamp":1615874346252388899,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I210316 05:59:06.257851 8451 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 508 ={"Timestamp":1615874346254986792,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","User":"root","DescriptorID":51,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I210316 05:59:06.270433 7076 server/server_sql.go:840  [n1] 509  done ensuring all necessary startup migrations have run
I210316 05:59:06.270665 7076 1@server/server.go:2061  [n1] 510  serving sql connections
    sqlsmith_test.go:70: seed: -185824220450351350
    sqlsmith_test.go:82: setup: rand-tables
    sqlsmith_test.go:92: setting: no-mutations+rand [disable LIMIT compare mode(disable mutations, disable impure funcs, ignore fns: "^crdb_internal", ignore fns: "^version", disable LIMIT, output sort) simple datums postgres mode(compare mode(disable mutations, disable impure funcs, ignore fns: "^crdb_internal", ignore fns: "^version", disable LIMIT, output sort), disable WITH, simple datums, ignore fns: "^current_", postgres, ignore fns: "^sha", ignore fns: "^quote", ignore fns: "^pg_collation_for") disable window funcs avoid consts disable WITH disable mutations]
I210316 05:59:06.271505 8626 jobs/job_scheduler.go:360  [n1] 511  waiting 2m0s before scheduled jobs daemon start
    panic.go:969: unimplemented: unable to encode JSON as a table key
        goroutine 7076 [running]:
        runtime/debug.Stack(0xc003906c98, 0x47cd8a0, 0xc001db5080)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x278
        panic(0x47cd8a0, 0xc001db5080)
        	/usr/local/go/src/runtime/panic.go:969 +0x1b9
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc001782000, 0x5a89640, 0xc000074110)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x31e
        panic(0x47cd8a0, 0xc001db5080)
        	/usr/local/go/src/runtime/panic.go:969 +0x1b9
        github.com/cockroachdb/cockroach/pkg/sql/mutations.statisticsMutator.func1(0xc0023c26c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/mutations/mutations.go:222 +0x668
        github.com/cockroachdb/cockroach/pkg/sql/mutations.statisticsMutator(0xc001db4720, 0xc001db4750, 0x3, 0x3, 0x0, 0x0, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/mutations/mutations.go:285 +0xbe6
        github.com/cockroachdb/cockroach/pkg/sql/mutations.MultiStatementMutation.Mutate(0x533a380, 0xc001db4720, 0xc001db4750, 0x3, 0x3, 0x0, 0xc002958b40, 0x7ffffffc000000e0, 0xc003907700)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/mutations/mutations.go:89 +0x4e
        github.com/cockroachdb/cockroach/pkg/sql/rowenc.RandCreateTables(0xc001db4720, 0x4c43a49, 0x5, 0x3, 0xc0039077f0, 0x3, 0x3, 0xe0, 0x4c0b9c, 0x7ac34c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowenc/testutils.go:1105 +0x224
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc001db4720, 0xc003907de8, 0x3)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:71 +0x1be
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.TestGenerateParse(0xc000d02180)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith_test.go:93 +0x4d3
        testing.tRunner(0xc000d02180, 0x5338058)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
--- FAIL: TestGenerateParse (0.11s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestGenerateParse PKG=./pkg/internal/sqlsmith TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestGenerateParse.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(internal/sqlsmith).TestGenerateParse failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2782309&tab=buildLog) on [master@98ae00235120443042f0d19565e47c237ed21dab](https://github.com/cockroachdb/cockroach/commits/98ae00235120443042f0d19565e47c237ed21dab):

```
I210316 14:25:48.689653 7897 sql/temporary_schema.go:531  [n1] 478  completed temporary object cleanup job
I210316 14:25:48.689745 7897 sql/temporary_schema.go:610  [n1] 479  temporary object cleaner next scheduled to run at 2021-03-16 14:55:48.685034098 +0000 UTC
I210316 14:25:48.692004 7987 sql/sqlliveness/slstorage/slstorage.go:352  [n1] 480  inserted sqlliveness session 4df5804198d7435dbec2cbcfed2ef0c1
I210316 14:25:48.692071 7987 sql/sqlliveness/slinstance/slinstance.go:144  [n1] 481  created new SQL liveness session 4df5804198d7435dbec2cbcfed2ef0c1
I210316 14:25:48.699255 7951 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 482 ={"Timestamp":1615904748684401835,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I210316 14:25:48.711433 8111 migration/migrationmanager/manager.go:102  [n1,intExec=set-setting,migration-mgr] 483  no need to migrate, cluster already at newest version
I210316 14:25:48.723688 8111 util/log/event_log.go:32  [n1,intExec=set-setting] 484 ={"Timestamp":1615904748704953451,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'20.2-48'"],"SettingName":"version","Value":"20.2-48"}
I210316 14:25:48.732341 7780 3@vendor/github.com/cockroachdb/pebble/db.go:1449  [n1,pebble] 485  [JOB 3] WAL created 000004
I210316 14:25:48.788413 8416 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 486 ={"Timestamp":1615904748753377657,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"83371182-72b4-4b8b-a7cc-37ff71b060ec"}
I210316 14:25:48.798749 8484 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 487 ={"Timestamp":1615904748792530564,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I210316 14:25:48.805066 8293 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 488 ={"Timestamp":1615904748799024794,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","User":"root","DescriptorID":51,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I210316 14:25:48.839796 7277 server/server_sql.go:840  [n1] 489  done ensuring all necessary startup migrations have run
I210316 14:25:48.840068 7277 1@server/server.go:2061  [n1] 490  serving sql connections
I210316 14:25:48.840105 8742 jobs/job_scheduler.go:360  [n1] 491  waiting 4m0s before scheduled jobs daemon start
I210316 14:25:48.844830 8744 server/auto_upgrade.go:55  [n1] 492  no need to upgrade, cluster already at the newest version
    sqlsmith_test.go:70: seed: -310824796787301440
    sqlsmith_test.go:82: setup: rand-tables
    sqlsmith_test.go:92: setting: no-ddl+rand [disable LIMIT disable WITH postgres mode(compare mode(disable mutations, disable impure funcs, ignore fns: "^crdb_internal", ignore fns: "^version", disable LIMIT, output sort), disable WITH, simple datums, ignore fns: "^current_", postgres, ignore fns: "^sha", ignore fns: "^quote", ignore fns: "^pg_collation_for") disable DDLs]
W210316 14:25:48.848094 8026 kv/kvserver/intentresolver/intent_resolver.go:758  [-] 493  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
W210316 14:25:48.848318 7988 jobs/registry.go:714  [-] 494  canceling all adopted jobs due to stopper quiescing
W210316 14:25:48.848732 7987 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 495  exiting heartbeat loop
    panic.go:969: unimplemented: unable to encode JSON as a table key
        goroutine 7277 [running]:
        runtime/debug.Stack(0xc004bcec98, 0x47cf920, 0xc002676690)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:110 +0x278
        panic(0x47cf920, 0xc002676690)
        	/usr/local/go/src/runtime/panic.go:969 +0x1b9
        github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc0020ca000, 0x5a8bd00, 0xc000072120)
        	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x31e
        panic(0x47cf920, 0xc002676690)
        	/usr/local/go/src/runtime/panic.go:969 +0x1b9
        github.com/cockroachdb/cockroach/pkg/sql/mutations.statisticsMutator.func1(0xc002e398c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/mutations/mutations.go:222 +0x668
        github.com/cockroachdb/cockroach/pkg/sql/mutations.statisticsMutator(0xc0025680f0, 0xc004858370, 0x5, 0x5, 0x0, 0x0, 0x0, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/mutations/mutations.go:285 +0xbe6
        github.com/cockroachdb/cockroach/pkg/sql/mutations.MultiStatementMutation.Mutate(0x533c658, 0xc0025680f0, 0xc004858370, 0x5, 0x5, 0x0, 0xc0047f9cc0, 0x7ffffffc000000e0, 0xc004bcf700)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/mutations/mutations.go:89 +0x4e
        github.com/cockroachdb/cockroach/pkg/sql/rowenc.RandCreateTables(0xc0025680f0, 0x4c45ac9, 0x5, 0x5, 0xc004bcf7f0, 0x3, 0x3, 0xe0, 0x4c0b9c, 0x7ac54c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowenc/testutils.go:1105 +0x224
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.randTables(0xc0025680f0, 0xc004bcfde8, 0x3)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/setup.go:71 +0x1be
        github.com/cockroachdb/cockroach/pkg/internal/sqlsmith.TestGenerateParse(0xc0004bfb00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/internal/sqlsmith/sqlsmith_test.go:93 +0x4d3
        testing.tRunner(0xc0004bfb00, 0x533a330)
        	/usr/local/go/src/testing/testing.go:1123 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1168 +0x2b3
--- FAIL: TestGenerateParse (0.30s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestGenerateParse PKG=./pkg/internal/sqlsmith TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestGenerateParse.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/mutations/mutations.go
+++ b/pkg/sql/mutations/mutations.go
@@ -210,6 +210,10 @@ func statisticsMutator(
 			n := rng.Intn(10)
 			seen := map[string]bool{}
 			colType := tree.MustBeStaticallyKnownType(col.Type)
+			// The JSON family does not have a key encoding.
+			if colType.Family() == types.JsonFamily {
+				return
+			}
 			h := stats.HistogramData{
 				ColumnType: colType,
 			}

